builders/txBuilder.go

```diff
@@ -40,52 +41,103 @@ func NewTxBuilder(txSigner TxSigner) (*txBuilder, error) {
 // createTransaction assembles a transaction from the provided arguments
 func (builder *txBuilder) createTransaction(arg data.ArgCreateTransaction) *data.Transaction {
 	return &data.Transaction{
-		Nonce:     arg.Nonce,
-		Value:     arg.Value,
-		RcvAddr:   arg.RcvAddr,
-		SndAddr:   arg.SndAddr,
-		GasPrice:  arg.GasPrice,
-		GasLimit:  arg.GasLimit,
-		Data:      arg.Data,
-		Signature: arg.Signature,
-		ChainID:   arg.ChainID,
-		Version:   arg.Version,
-		Options:   arg.Options,
+		Nonce:             arg.Nonce,
+		Value:             arg.Value,
+		RcvAddr:           arg.RcvAddr,
+		SndAddr:           arg.SndAddr,
+		GasPrice:          arg.GasPrice,
+		GasLimit:          arg.GasLimit,
+		Data:              arg.Data,
+		Signature:         arg.Signature,
+		ChainID:           arg.ChainID,
+		Version:           arg.Version,
+		Options:           arg.Options,
+		GuardianAddr:      arg.GuardianAddr,
+		GuardianSignature: arg.GuardianSignature,
 	}
 }
 
-// ApplySignatureAndGenerateTx will apply the corresponding sender and compute the signature field and
+// ApplyUserSignatureAndGenerateTx will apply the corresponding sender and compute the signature field and
 // generate the transaction instance
-func (builder *txBuilder) ApplySignatureAndGenerateTx(
+func (builder *txBuilder) ApplyUserSignatureAndGenerateTx(
 	skBytes []byte,
 	arg data.ArgCreateTransaction,
 ) (*data.Transaction, error) {
-
+	newArg := arg
```

Comment:
why don`t we use arg directly and make a copy ?

Reply:
removed.